pennylane/qchem/openfermion_obs.py

```diff
@@ -649,6 +658,169 @@ def dipole_of(
+        active_electrons (int): Number of active electrons. If not specified, all electrons
+            are considered to be active.
+        active_orbitals (int): Number of active orbitals. If not specified, all orbitals
+            are considered to be active.
+        mapping (str): transformation used to map the fermionic Hamiltonian to the qubit Hamiltonian. Input values can be ``'jordan_wigner'``, ``'parity'`` or ``'bravyi_kitaev'``.
+        outpath (str): path to the directory containing output files
+        wires (Wires, list, tuple, dict): Custom wire mapping for connecting to Pennylane ansatz.
+            For types ``Wires``/``list``/``tuple``, each item in the iterable represents a wire label
+            corresponding to the qubit number equal to its index.
+            For type dict, only int-keyed dict (for qubit-to-wire conversion) is accepted for
+            partial mapping. If None, will use identity map.
+        args (array[array[float]]): initial values of the differentiable parameters
+        cutoff (float): Cutoff value for including the matrix elements
+            :math:`\langle \alpha \vert \hat{{\bf r}} \vert \beta \rangle`. The matrix elements
+            with absolute value less than ``cutoff`` are neglected.
+
+    Returns:
+        list[pennylane.Hamiltonian]: the qubit observables corresponding to the components
+        :math:`\hat{D}_x`, :math:`\hat{D}_y` and :math:`\hat{D}_z` of the dipole operator.
+
+
+    """
+
+    if method not in ["dhf", "openfermion"]:
+        raise ValueError("Only 'dhf', and 'openfermion' backends are supported.")
+
+    if mapping.strip().lower() not in ["jordan_wigner", "parity", "bravyi_kitaev"]:
+        raise ValueError(
+            f"'{mapping}' is not supported."
+            f"Please set the mapping to 'jordan_wigner', 'parity' or 'bravyi_kitaev'."
+        )
+
+    symbols = molecule.symbols
+    coordinates = molecule.coordinates
+
+    if len(coordinates) == len(symbols) * 3:
+        geometry_dhf = qml.numpy.array(coordinates.reshape(len(symbols), 3))
+        geometry_hf = coordinates
+    elif len(coordinates) == len(symbols):
+        geometry_dhf = qml.numpy.array(coordinates)
+        geometry_hf = coordinates.flatten()
+
+    if molecule.mult != 1:
+        raise ValueError(
+            "Open-shell systems are not supported. Change the charge or spin multiplicity of the molecule."
+        )
+
+    wires_map = None
+
+    if wires:
+        wires_new = qml.qchem.convert._process_wires(wires)
+        wires_map = dict(zip(range(len(wires_new)), list(wires_new.labels)))
+
+    core, active = qml.qchem.active_space(
+        molecule.n_electrons, molecule.n_orbitals, molecule.mult, active_electrons, active_orbitals
+    )
+
+    if method == "dhf":
+
+        if args is None and isinstance(geometry_dhf, qml.numpy.tensor):
+            geometry_dhf.requires_grad = False
+        mol = qml.qchem.Molecule(
```

Comment:
On a second thought, do we need to make the whole `Moelcule` again or should we add the processing regarding the coordinates to the molecule class?

Reply:
This changes the basis_set and r as well, depending on whether gradients are required for coordinates or not.